core/src/main/scala/org/apache/spark/rdd/AsyncRDDActions.scala

```diff
@@ -61,6 +62,36 @@ class AsyncRDDActions[T: ClassTag](self: RDD[T]) extends Serializable with Loggi
       (index, data) => results(index) = data, results.flatten.toSeq)
   }
 
+
+  /**
+   * Returns a future of an aggregation across the RDD.
+   *
+   * @see [[RDD.aggregate]] which is the synchronous version of this method.
+   */
+  def aggregateAsync[U](zeroValue: U)(seqOp: (U, T) => U, combOp: (U, U) => U): FutureAction[U] =
+    self.withScope {
```

Comment:
In the synchronous version of `aggregate`, the `zeroValue` is cloned, which requires adding an implicit `ClassTag[U]` argument. I didn't really understand the motivation for that, so I didn't do it here, but I was hoping that someone who understood the cloning could let me know here.

Reply:
IMHO one reason could be the rule of fail fast and early. As the cloning uses serialisation/deserialisation which anyway needed for sending the neutral element to the executors this way serialisation of the neutral element is tested when the operator is specified not during the lazy execution in the middle of a long chain of operations.